### ISSUE TITLE
Unexpected result when including file from inside .macro

### ISSUE THREAD

**Issue by ReallyNiceGuy** [cc65/cc65#1473]
When using .include inside a .macro, the contents of the included file are added after the macro expansion, no matter where the .include was done.

Ex:
```
.org $080D
.segment "STARTUP"
.segment "INIT"
.segment "ONCE"
.segment "CODE"

.macro LITERAL2 arg
    .pushcharmap
    .include "ascii_charmap.inc"
    .byte arg
    .popcharmap
.endmacro

LITERAL2 "text in latin1"
```
This results in:
```
ca65 V2.19 - Git f272bc8f4
Main file   : min.asm
Current file: min.asm

000000r 1               .org $080D
00080D  1               .segment "STARTUP"
00080D  1               .segment "INIT"
00080D  1               .segment "ONCE"
00080D  1               .segment "CODE"
00080D  1    
00080D  1    
00080D  1               .macro LITERAL2 arg 
00080D  1                   .pushcharmap
00080D  1                   .include "ascii_charmap.inc"
00080D  1                   .byte arg 
00080D  1                   .popcharmap
00080D  1               .endmacro
00080D  1    
```
**NOTE: this is still in PETSCII**
```
00080D  1  CC 4F 52 45  LITERAL2  {"Lorem Ipsum",0}
000811  1  4D 20 C9 50  
000815  1  53 55 4D 00  
000819  1    
000819  2               ;/*****************************************************************************/
000819  2               ;/*                                                                           */
000819  2               ;/*                              ascii_charmap.inc                            */
000819  2               ;/*                                                                           */
...
000819  2               ;/*****************************************************************************/
000819  2
000819  2               ;/* ASCII */
000819  2               .charmap $00, $00
000819  2               .charmap $01, $01
...
000819  2               .charmap $FD, $FD
000819  2               .charmap $FE, $FE
000819  2               .charmap $FF, $FF
000819  2
000819  1
```
**NOTE: Now we are in latin1, but we expected to be in PETSCII because of the .pushcharmap and .popcharmap pair.**

**Comment by greg-king5** [comment]
This issue is related to issue #231.

**Comment by ReallyNiceGuy** [comment]
My patch will not fix issue #231. It will make it fail to compile, as the include file is processed at the time of the definition, not at the time of the expansion.

**Comment by oliverschmidt** [comment]
Fixed with https://github.com/cc65/cc65/pull/1477.
